network/unix_domain_server.h

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2025 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+
+#include <atomic>
+#include <condition_variable>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <thread>
+
+#include "messages.h"
+#include "message_handler.h"
+
+namespace Network
+{
+
+// Default message handler if user doesn't provide/implement one.
+class DefaultMessageHandler : public IMessageHandler
+{
+public:
+    DefaultMessageHandler();
+    void OnConnect() override;
+    void HandleMessage(std::unique_ptr<ISerializable> message,
+                       SocketConnection*              client_conn) override;
+    void OnDisconnect() override;
+};
+
+class UnixDomainServer
+{
+public:
+    // Constructs the server, taking ownership of the provided IMessageHandler.
+    explicit UnixDomainServer(
```

Comment:
The constructor's default argument for handler creates a null pointer, which will cause a crash if the server is initialized without a specific handler. Maybe better with something like this?
```
explicit UnixDomainServer(
    std::unique_ptr<IMessageHandler> handler = std::make_unique<DefaultMessageHandler>());
    ```